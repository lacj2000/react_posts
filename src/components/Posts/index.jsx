import P from 'prop-types';
import Post from '../Post';
import './styles.css';

const Posts = ({ posts = [] }) => (
  <div className="posts">
    {posts.map((post) => (
      <Post key={post.id} title={post.title} body={post.body} url={post.url} />
    ))}
  </div>
);

Posts.defaultProps = {
  posts: [],
};

Posts.propTypes = {
  posts: P.arrayOf(
    P.shape({
      url: P.string.isRequired,
      title: P.string.isRequired,
      body: P.string.isRequired,
    }),
  ),
};

export default Posts;
