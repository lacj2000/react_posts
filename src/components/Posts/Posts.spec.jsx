import { render, screen } from '@testing-library/react';
import Posts from '.';

const props = {
  posts: [
    {
      id: 1,
      title: 'title 01',
      body: 'body 01',
      url: 'img/test01.png',
    },
    {
      id: 2,
      title: 'title 02',
      body: 'body 02',
      url: 'img/test02.png',
    },
    {
      id: 3,
      title: 'title 03',
      body: 'body 03',
      url: 'img/test03.png',
    },
  ],
};

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByAltText(/title/i)).toHaveLength(3);

    expect(screen.getByRole('img', { name: /title 01/i })).toHaveAttribute('src', 'img/test01.png');
  });

  it('must check Post component with empty posts parameter', () => {
    render(<Posts />);

    expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
