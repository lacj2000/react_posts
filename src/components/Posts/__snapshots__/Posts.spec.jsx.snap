// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 01"
      src="img/test01.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 01
      </h2>
      <p>
        body 01
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 02"
      src="img/test02.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 02
      </h2>
      <p>
        body 02
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 03"
      src="img/test03.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 03
      </h2>
      <p>
        body 03
      </p>
    </div>
  </div>
</div>
`;
